apps/chat/conversation.py

```diff
@@ -192,69 +201,71 @@ def _compress_chat_history(
     )
     return history, last_message, summary
 
+def truncate_tokens(history, max_token_limit, llm, input_message_tokens):
+    """Removes old messages until the token count is below the max limit."""
+    pruned_memory = []
+    history_tokens = llm.get_num_tokens_from_messages(history)
+    while history and history_tokens + input_message_tokens > max_token_limit:
+        pruned_memory.append(history.pop(0))
+        history_tokens = llm.get_num_tokens_from_messages(history)
+    return history, pruned_memory
 
-def compress_chat_history_from_messages(
-    llm, history, keep_history_len: int, max_token_limit: int, input_messages: list
-):
-    summary = history.pop(0).content if history[0].type == ChatMessageType.SYSTEM else None
-    history, pruned_memory = history[-keep_history_len:], history[:-keep_history_len]
-    # The latest message
-    latest_message = history[-1]
 
+def summarize_history(llm, history, max_token_limit, input_message_tokens, summary):
+    pruned_memory = []
+    history_tokens = llm.get_num_tokens_from_messages(history)
     summary_tokens = (
         llm.get_num_tokens_from_messages([SystemMessage(content=summary)])
         if summary
         else INITIAL_SUMMARY_TOKENS_ESTIMATE
     )
-    input_message_tokens = llm.get_num_tokens_from_messages(input_messages)
-    history_tokens = llm.get_num_tokens_from_messages(history)
-    first_pass_done = False  # ensure we do at least one loop
+    while history and history_tokens + summary_tokens + input_message_tokens > max_token_limit:
+        pruned_memory.append(history.pop(0))
+        history_tokens = llm.get_num_tokens_from_messages(history)
+    summary = _get_new_summary(llm, pruned_memory, summary, max_token_limit)
+    return history, pruned_memory, summary
 
-    while (
-        not first_pass_done
-        # Check `_tokens_exceeds_limit` here, since the generated summary can cause a token limit breach
-        or _tokens_exceeds_limit(
-            history, token_count=(history_tokens + summary_tokens + input_message_tokens), limit=max_token_limit
+def compress_chat_history_from_messages(
+    llm, history, keep_history_len: int, max_token_limit: int, input_messages: list, history_mode: str = None
+):
+    """
+    Handles chat history compression based on selected mode:
+    - "Summarize": Summarizes older messages when exceeding token limit.
+    - "Truncate Tokens": Deletes old messages when exceeding token limit.
+    - "Max History Length": Always keeps the last `keep_history_len` messages.
+    """
+    summary = history.pop(0).content if history and history[0].type == ChatMessageType.SYSTEM else None
+    history, pruned_memory = history[-keep_history_len:], history[:-keep_history_len]
+    latest_message = history[-1] if history else None
+    input_message_tokens = llm.get_num_tokens_from_messages(input_messages)
+    if history_mode == "Max History Length":
```

Comment:
With some of the other suggestions you should be able to use the `PipelineChatHistoryModes` enum:

```suggestion
    if history_mode == PipelineChatHistoryModes.MAX_HISTORY_LENGTH:
```